### DISCUSSION TITLE
Get the length of an attribute value (without the slow strlen)

### DISCUSSION THREAD

**Discussion by muratmat** [zeux/pugixml#405] [Q&A] [answered]
Would it be possible to get the length of an attribute value?
In other words, would pugixml be able to calculate the length of an attribute value without calling a (slow)

`strlen(pugi::xml_attribute.value()) ?`

I'm thinking about on how to get the length of base64 encoded data within an svg file.
For example:

`<image width="398" height="1194" x="2" y="-400" xlink:href="data:image/png;charset=utf-8;base64,iVBORw0KG......"/>`

Many thanks in advance!

**Comment by zeux** [comment, marked as answer]
pugixml stores all values as zero-terminated strings so strlen is the only option if you do need the length.

Note that to decode the BASE64 contents you don't necessarily need the length - you can just parse the string character by character until reaching the null terminator.
